### ISSUE TITLE
MPV 3D Playback on Lume Pad 2

### ISSUE THREAD

**Issue by cryptochainer** [jakedowns/mpv-android#3]
While the player loads, on Lume Pad 2, it does not properly trigger the 3D effect.  There are two overlaid images, as expected, but the device's 3D mode is not triggered.  I would imagine it would not be terribly difficult to update the application, using the SDK data available from Leia.  I have made a donation to this cause, in hope that this can be done sometime very soon!

**Comment by 205rallye** [comment]
I have the same problem...

**Comment by jakedowns** [comment]
@cryptochainer , @205rallye hey there, I'm working on this now and will post back when it's ready
thanks for the support!

**Comment by cryptochainer** [comment]
Awesome thanks very much. On May 6, 2023, at 9:43 AM, Jake Downs ***@***.***> wrote:﻿
@cryptochainer , @205rallye hey there, I'm working on this now and will post back when it's ready
thanks for the support!

**Comment by 205rallye** [comment]
Can you make the option of view 3D up & down too??? Or is limited to SBS for any reason that I don't know?

Thank you for your efforts.

**Comment by jakedowns** [comment]
@cryptochainer got this working locally, gonna work on putting together a release APK

@205rallye you bet! i'll make sure that's included as an option

basic proof of concept apk here: https://github.com/jakedowns/mpv-android/releases/tag/v2023_05_08

I still need to add OU support back in, and fix a couple other things. but basic SBS playback is working now (You need to go to the settings gear icon in the top right during playback and then tap "3D SBS")

**Comment by jakedowns** [comment]
@205rallye Over Under support added back, but it's slightly broken. working on a fix

https://github.com/jakedowns/mpv-android/releases/tag/v2023_05_09_over_under_support
![Screenshot_20230509_175649](https://github.com/jakedowns/mpv-android/assets/1683122/d1f8f046-5632-4fb5-9392-dd1c9356fa0e)

**Comment by cryptochainer** [comment]
I would like to see an option to switch L-R,  in SBS, if that's possible...

**Comment by jakedowns** [comment]
oh there is a "Swap" option but i think i broke it with my last update. ill
fix that asap

**Comment by jakedowns** [comment]
> I would like to see an option to switch L-R, in SBS, if that's possible...

This should work: 

https://github.com/jakedowns/mpv-android/releases/tag/v2023_05_09_b_fix_sbs_swap

**Comment by cryptochainer** [comment]
One additional suggestion. On set-top media players, many will recognize an extension/flag, and automatically enable the correct mode.  For example, AwesomeMovie.3DSBS.mkv, or AwesomeMovie.3DTAB.mkv, will trigger the proper 3D mode on the device.  It would be cool to have this happen, and save me two dreaded clicks! ;)

**Comment by jakedowns** [comment]
this is something i'm working on adding to my Into3D video player for Nreal Air glasses,
so i will definitley add this feature to my MPV fork soon 🙌

Added in a new Issue Ticket here: https://github.com/jakedowns/mpv-android/issues/4

**Comment by cryptochainer** [comment]
Shockingly, I have the Nreal glasses too.  Does the 3D app currently function?  I also have the Virtue ones, which I don't think 3D will work properly with, without the neckband.

**Comment by jakedowns** [comment]
> Shockingly, I have the Nreal glasses too. Does the 3D app currently function? I also have the Virtue ones, which I don't think 3D will work properly with, without the neckband.

yup https://play.google.com/store/apps/details?id=com.jakedowns.into3D

**Comment by 205rallye** [comment]
Is there any possibility for playing photo files like *.mpo or *.jps?

**Comment by jakedowns** [comment]
@205rallye i can certainly look into it. I've added a separate issue for that request

**Comment by 205rallye** [comment]
> @205rallye i can certainly look into it. I've added a separate issue for that request

It'll be great. Thank you for all your efforts...

**Comment by HuangArmagh** [comment]
I'm experiencing a subtitle display issue with the MPV 3D player on Lume Pad 2. Interestingly, the subtitles work fine on Lume Pad 1, but on Lume Pad 2, both the position and size of the subtitles are incorrect. 

I've attached two screenshots for a clearer understanding – one showing the subtitle display on Lume Pad 2 and the other on Lume Pad 1 for comparison.

Thank you very much for your work, and I look forward to your response and a resolution to this issue. I am happy to provide more information if needed.

lumepad2
![Screenshot_20231215_045809](https://github.com/jakedowns/mpv-android/assets/31194798/47ff2ca2-eed7-4785-a2a7-5d950f8598d7)


lumepad1
![2023_12_15_21 31 20](https://github.com/jakedowns/mpv-android/assets/31194798/e138d2df-bb2e-4674-969e-ee8a197265b6)
